speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -327,6 +335,53 @@ static void OnActivate(void* /* dummy */)
     LoadDefaultSettings();
 }
 
+/// @brief Opens a file dialog and changes the button to reflect this choice.
+static void SelectFile(void* /* dummy */)
```

Comment:
Just like gerrie said, please add white space to the function, place all statements on their own line. The only exception is `if(something) return;` but that's not the case here, since you are invoking more statements before the retuns, so please reformat that to place them all on newlines. Group blocks of related logic together, but then add a whitespace after that for readability.